### PR TITLE
Use relative path, include tag in name and folder

### PR DESCRIPTION
**Pull request by manics**
- Don't use `/opt` as the root of the archive as this makes it difficult to relocate. For example, if you wanted to download multiple versions and symlink just one version you would end up with paths like `/opt/ice/opt/Ice-3.6.5`
- Include the tag in the folder name which should make it easier to see which build is installed if this is updated in future
- Lowercase name

Tag: `0.3.0`
E.g. output would be `ice-3.6.5-0.3.0-ubuntu1804-amd64.tar.gz` containing paths `ice-3.6.5-0.3.0/`

It'd be nice if the wheel could be renamed too, but that prevents it being installed.

When everyone is happy with the layout I'll look into a CentOS-8 build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ deploy:
   - provider: releases
     api_key: $GITHUB_OAUTH_TOKEN
     file_glob: true
-    file: dist/Ice*
+    file: dist/ice*
     skip_cleanup: true
     on:
       tags: true
```
Comment (jburel) [inline .travis.yml]: Add extra line

Comment (manics) [inline .travis.yml]:
```suggestion
      tags: true

```

**Comment by jburel** [comment]
Adjustment made to omero-install to test this PR
Built and tagged and pushed to my account
https://travis-ci.org/jburel/omero-install/builds/617916225 is green

File: build.sh [hunk 2]
```diff
@@ -3,17 +3,21 @@
 set -eux
 
 ICE_VERSION=${1:-"3.6.5"}
+BUILD=${TRAVIS_BRANCH:-dev}
```
Comment (jburel) [inline build.sh:6]: what is the rationale for adding travis-branch in the name?
When we tag, we will need to remember to pass a value, so we do not end with ``-dev`` in the name.
This should be indicated in the README

Comment (jburel) [reply build.sh:6]: I have not tested the tagging and name change during the tagging

**Comment by jburel** [comment]
The modification of the name is a breaking change so all components using it will need to be adjusted

File: build.sh (hunk 2 above)
Comment (manics) [reply build.sh:6]: It was fixed in the last commit after you tested

**Comment by manics** [comment]
Ideally we'd have this packaged in a repository (e.g. ppa) which would take care of versioning, but in the absence of that I think we should include the build-tag or some other version indicator so it's clear what's installed.

**Comment by jburel** [comment]
@manics understood.
I need to test the "tagging" step i.e. will the name of the tag be correctly picked?
Did you check that?

**Comment by jburel** [comment]
https://github.com/manics/zeroc-ice-ubuntu1804/releases is looking good

**Comment by jburel** [comment]
Happy with layout
If no vetoes, I think we should merge and tag.

**Pull request merged**